src/wagtailmarkdown/static/wagtailmarkdown/js/easymde-controller.js

```diff
@@ -1,12 +1,11 @@
-// wagtailmarkdown/static/wagtailmarkdown/js/easymde-controller.js
-
 class EasyMDEContainer extends window.StimulusModule.Controller {
-    static values = { autodownload: Boolean };
+    static values = { autodownload: Boolean};
 
     connect() {
-        if (this.autodownloadValue != null) {
+        if (this.hasAutodownloadValue) {
```

Comment:
tl;dr - EasyMDE will try to detect if FA is present if initialized with a null value. Given `static values = { autodownload: Boolean};` a missing value would be false, so we were initialising EasyMDE with "don't download FA"

Reply:
Sorry. Just reading this now. 

The `hasSomeValue` just checks if the value is declared in the DOM so I'm not sure this is what you want. 

```
<textarea data-controller="easymde" ... (No html data attribute for the value )
// this.hasAutodownloadValue - will be false 
<textarea data-controller="easymde" data-easymde-autodownload="false" ...
// this.hasAutodownloadValue - will be true
<textarea data-controller="easymde"  data-easymde-autodownload="true" ...
// this.hasAutodownloadValue - will be true 
```

I think the issue was the confusion of checking against null with a `!=` as this will try to coerce the Boolean into a comparable value. As a general guide it's good to use strict inequality and try to align checks with the type set of the value. 

Or keep the code simpler with returns or a non-comparison approach.

```
easymdeAttach(this.element.id, this.autodownloadValue);
```

`this.autodownloadValue` will always return a Boolean, Stimulus does this for us, so why do any branch logic at all. The default will be `false` if not set in the html.

Reply:
https://stimulus.hotwired.dev/reference/values#properties-and-attributes

Reply:
Thank you for additional context, LB.

`this.hasAutodownloadValue` is what I want because we only set the data attribute the setting is present, hence I only want to pass it down when the attribute is set.

https://github.com/Ionaru/easy-markdown-editor?tab=readme-ov-file#configuration

> autoDownloadFontAwesome: If set to true, force downloads Font Awesome (used for icons). If set to false, prevents downloading. Defaults to undefined, which will intelligently check whether Font Awesome has already been included, then download accordingly.

Reply:
Ok. No problems.